recipes/halide/build.sh

```diff
@@ -0,0 +1,16 @@
+if [[ $(uname) == "Darwin" ]]; then
+    # don't use clang++ from the env
+    export CXX=c++
```

Comment:
Interesting. Maybe we should discuss the pros and cons of this choice as it pertains to Halide.